### ISSUE TITLE
failed to run rootfull container when trying to mount device block

### ISSUE THREAD

**Issue by tupyy** [containers/podman#14450]
/kind bug

**Description**

Running rootfull container when trying to mount block device results in error.

**Steps to reproduce the issue:**

1. Create a loop device
```
[root@fedora cosmin]# ls -al /dev | grep ^b
brw-rw----   1 root   disk    253,     0 Jun  1 22:15 dm-0
brw-rw----   1 root   disk      7,     1 Jun  1 22:29 loop1
brw-rw----   1 root   disk    259,     0 Jun  1 22:15 nvme0n1
brw-rw----   1 root   disk    259,     1 Jun  1 22:15 nvme0n1p1
brw-rw----   1 root   disk    259,     2 Jun  1 22:15 nvme0n1p2
brw-rw----   1 root   disk    259,     3 Jun  1 22:15 nvme0n1p3
brw-rw----   1 root   disk      8,     0 Jun  1 22:15 sda
brw-rw----   1 root   disk      8,     1 Jun  1 22:15 sda1
brw-rw----   1 root   disk    252,     0 Jun  1 22:15 zram0
```
2.  Execute:
```
podman run -it --rm --privileged --mount=type=bind,src=/dev/loop1,dst=/mnt --log-level=trace --security-opt label=disable fedora:35 /bin/bash
```


**Describe the results you received:**
```
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called run.PersistentPreRunE(podman run -it --rm --privileged --mount=type=bind,src=/dev/loop1,dst=/mnt --log-level=debug --security-opt label=disable fedora:35 /bin/bash) 
DEBU[0000] Merged system config "/usr/share/containers/containers.conf" 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /var/lib/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver                           
DEBU[0000] Using graph root /var/lib/containers/storage 
DEBU[0000] Using run root /run/containers/storage       
DEBU[0000] Using static dir /var/lib/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/libpod                    
DEBU[0000] Using volume path /var/lib/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] Cached value indicated that native-diff is usable 
DEBU[0000] backingFs=btrfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
INFO[0000] [graphdriver] using prior storage driver: overlay 
DEBU[0000] Initializing event backend journald          
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/crun"            
INFO[0000] Setting parallel job count to 37             
DEBU[0000] Pulling image fedora:35 (policy: missing)    
DEBU[0000] Looking up image "fedora:35" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/000-shortnames.conf" 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:35" ... 
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] Found image "fedora:35" as "registry.fedoraproject.org/fedora:35" in local containers storage 
DEBU[0000] Found image "fedora:35" as "registry.fedoraproject.org/fedora:35" in local containers storage ([overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora:35" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:35" ... 
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora:35" as "registry.fedoraproject.org/fedora:35" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora:35" as "registry.fedoraproject.org/fedora:35" in local containers storage ([overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240) 
DEBU[0000] Looking up image "fedora:35" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:35" ... 
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] Found image "fedora:35" as "registry.fedoraproject.org/fedora:35" in local containers storage 
DEBU[0000] Found image "fedora:35" as "registry.fedoraproject.org/fedora:35" in local containers storage ([overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240) 
DEBU[0000] Inspecting image 750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240 
DEBU[0000] exporting opaque data as blob "sha256:750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] exporting opaque data as blob "sha256:750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] exporting opaque data as blob "sha256:750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] Inspecting image 750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240 
DEBU[0000] Inspecting image 750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240 
DEBU[0000] Inspecting image 750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240 
DEBU[0000] using systemd mode: false                    
DEBU[0000] No hostname set; container's hostname will default to runtime default 
DEBU[0000] Loading seccomp profile from "/usr/share/containers/seccomp.json" 
DEBU[0000] Adding mount /proc                           
DEBU[0000] Adding mount /dev                            
DEBU[0000] Adding mount /dev/pts                        
DEBU[0000] Adding mount /dev/mqueue                     
DEBU[0000] Adding mount /sys                            
DEBU[0000] Adding mount /sys/fs/cgroup                  
DEBU[0000] Successfully loaded 1 networks               
DEBU[0000] Allocated lock 0 for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
DEBU[0000] parsed reference into "[overlay@/var/lib/containers/storage+/run/containers/storage]@750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] exporting opaque data as blob "sha256:750037c05cfe1857e16500167c7c217658e15eb9bc6283020cfb3524c93d1240" 
DEBU[0000] Cached value indicated that overlay is not supported 
DEBU[0000] Check for idmapped mounts support            
DEBU[0000] Created container "b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65" 
DEBU[0000] Container "b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65" has work directory "/var/lib/containers/storage/overlay-containers/b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65/userdata" 
DEBU[0000] Container "b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65" has run directory "/run/containers/storage/overlay-containers/b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65/userdata" 
DEBU[0000] Handling terminal attach                     
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] backingFs=btrfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Cached value indicated that volatile is being used                                    
DEBU[0000] overlay: mount_data=lowerdir=/var/lib/containers/storage/overlay/l/QJBF3BNZOTPBCWUMQ3UO74K3KE,upperdir=/var/lib/containers/storage/overlay/2219a52ca004accfe6ef865f3454b5c97a7cbcf5cec8dba32372f8176d7c0dc6/diff,workdir=/var/lib/containers/storage/overlay/2219a52ca004accfe6ef865f3454b5c97a7cbcf5cec8dba32372f8176d7c0dc6/work,,volatile 
DEBU[0000] Made network namespace at /run/netns/netns-255759c3-0a5d-2d4a-4a4c-bc73a04e3453 for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
DEBU[0000] Mounted container "b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65" at "/var/lib/containers/storage/overlay/2219a52ca004accfe6ef865f3454b5c97a7cbcf5cec8dba32372f8176d7c0dc6/merged" 
DEBU[0000] Created root filesystem for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 at /var/lib/containers/storage/overlay/2219a52ca004accfe6ef865f3454b5c97a7cbcf5cec8dba32372f8176d7c0dc6/merged 
[DEBUG netavark::network::validation] "Validating network namespace..."                                                                                                                                                          
                                                                       [DEBUG netavark::commands::setup] "Setting up..."
                                                                                                                        [INFO  netavark::firewall] Using iptables firewall driver
                                                                                                                                                                                 [DEBUG netavark::network::core_utils] Setting sysctl value for net.ipv4.ip_forward to 1
                                                                                                                                                                                                                                                                        [DEBUG netavark::commands::setup] Setting up network podman with driver bridge
                                                [DEBUG netavark::network::core] Container veth name: "eth0"
                                                                                                           [DEBUG netavark::network::core] Brige name: "podman0"
                                                                                                                                                                [DEBUG netavark::network::core] IP address for veth vector: [10.88.0.7/16]
                                                                                                                                                                                                                                          [DEBUG netavark::network::core] Gateway ip address vector: [10.88.0.1/16]
             [DEBUG netavark::network::core] Configured static up address for eth0
                                                                                  [DEBUG netavark::network::core] Container veth mac: "62:0e:7e:ad:00:c0"
                                                                                                                                                         [DEBUG netavark::firewall::varktables::helpers] chain NETAVARK-1D8721804F16F created on table nat
                                                                                                                                                                                                                                                          [DEBUG netavark::firewall::varktables::helpers] rule -d 10.88.0.0/16 -j ACCEPT exists on table nat and chain NETAVARK-1D8721804F16F
                                                                                       [DEBUG netavark::firewall::varktables::helpers] rule -d 10.88.0.0/16 -j ACCEPT created on table nat and chain NETAVARK-1D8721804F16F
                                                                                                                                                                                                                           [DEBUG netavark::firewall::varktables::helpers] rule ! -d 224.0.0.0/4 -j MASQUERADE exists on table nat and chain NETAVARK-1D8721804F16F
                                                             [DEBUG netavark::firewall::varktables::helpers] rule ! -d 224.0.0.0/4 -j MASQUERADE created on table nat and chain NETAVARK-1D8721804F16F
                                                                                                                                                                                                      [DEBUG netavark::firewall::varktables::helpers] rule -s 10.88.0.0/16 -j NETAVARK-1D8721804F16F exists on table nat and chain POSTROUTING
                                        [DEBUG netavark::firewall::varktables::helpers] rule -s 10.88.0.0/16 -j NETAVARK-1D8721804F16F created on table nat and chain POSTROUTING
                                                                                                                                                                                 [DEBUG netavark::firewall::varktables::helpers] chain NETAVARK_FORWARD exists on table filter
                                                                                                                                                                                                                                                                              [DEBUG netavark::firewall::varktables::helpers] chain NETAVARK_FORWARD exists on table filter
                                                                     [DEBUG netavark::firewall::varktables::helpers] rule -d 10.88.0.0/16 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT exists on table filter and chain NETAVARK_FORWARD
                                                                                                                                                                                                                                                [DEBUG netavark::firewall::varktables::helpers] rule -d 10.88.0.0/16 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT created on table filter and chain NETAVARK_FORWARD
                                                                                                                      [DEBUG netavark::firewall::varktables::helpers] rule -s 10.88.0.0/16 -j ACCEPT exists on table filter and chain NETAVARK_FORWARD
                                                                                                                                                                                                                                                      [DEBUG netavark::firewall::varktables::helpers] rule -s 10.88.0.0/16 -j ACCEPT created on table filter and chain NETAVARK_FORWARD
                                                                                 [DEBUG netavark::commands::setup] {
                                                                                                                            "podman": StatusBlock {
                                                                                                                                                               dns_search_domains: Some(
                                                                                                                                                                                                        [],
                                                                                                                                                                                                                       ),
                                                                                                                                                                                                                                     dns_server_ips: Some(
                                                                                                                                                                                                                                                                          [],
                                                                                                                                                                                                                                                                                         ),
                                                                                                                                                                                                                                                                                                       interfaces: Some(
                                  {
                                                       "eth0": NetInterface {
                                                                                                     mac_address: "62:0e:7e:ad:00:c0",
                                                                                                                                                              subnets: Some(
                                                                                                                                                                                                        [
                                                                                                                                                                                                                                         NetAddress {
                                                                                                                                                                                                                                                                                         gateway: Some(
                                         10.88.0.1,
                                                                                       ),
                                                                                                                             ipnet: 10.88.0.7/16,
                                                                                                                                                                                 },
                                                                                                                                                                                                               ],
                                                                                                                                                                                                                                         ),
                                                                                                                                                                                                                                                               },
                                                                                                                                                                                                                                                                                 },
                                                                                                                                                                                                                                                                                               ),
                                                                                                                                                                                                                                                                                                         },
         }
          [DEBUG netavark::commands::setup] "Setup complete"
                                                            DEBU[0000] Adding nameserver(s) from network status of '[]' 
DEBU[0000] Adding search domain(s) from network status of '[]'                                                          
DEBU[0000] /etc/system-fips does not exist on host, not mounting FIPS mode subscription 
DEBU[0000] Setting Cgroups for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 to machine.slice:libpod:b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
DEBU[0000] reading hooks from /usr/share/containers/oci/hooks.d                                                                                                                                    
DEBU[0000] Workdir "/" resolved to host path "/var/lib/containers/storage/overlay/2219a52ca004accfe6ef865f3454b5c97a7cbcf5cec8dba32372f8176d7c0dc6/merged" 
DEBU[0000] Created OCI spec for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 at /var/lib/containers/storage/overlay-containers/b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65/userdata/config.json 
DEBU[0000] /usr/bin/conmon messages will be logged to syslog                                                                                                                                                                                       
DEBU[0000] running conmon: /usr/bin/conmon               args="[--api-version 1 -c b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 -u b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 -r /usr/bin/crun -b /var/lib/containers/storage/overlay-containers/b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65/userdata -p /run/containers/storage/overlay-containers/b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65/userdata/pidfile -n elated_bhaskara --exit-dir /run/libpod/exits --full-attach -s -l journald --log-level debug --syslog -t --conmon-pidfile /run/containers/storage/overlay-containers/b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /var/lib/containers/storage --exit-command-arg --runroot --exit-command-arg /run/containers/storage --exit-command-arg --log-level --exit-command-arg debug --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/libpod --exit-command-arg --network-config-dir --exit-command-arg  --exit-command-arg --network-backend --exit-command-arg netavark --exit-command-arg --volumepath --exit-command-arg /var/lib/containers/storage/volumes --exit-command-arg --runtime --exit-command-arg crun --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg --syslog --exit-command-arg container --exit-command-arg cleanup --exit-command-arg --rm --exit-command-arg b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65]"
INFO[0000] Running conmon under slice machine.slice and unitName libpod-conmon-b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65.scope 
DEBU[0000] Received: -1                                                                                                                               
DEBU[0000] Cleaning up container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
DEBU[0000] Tearing down network namespace at /run/netns/netns-255759c3-0a5d-2d4a-4a4c-bc73a04e3453 for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
[DEBUG netavark::commands::teardown] "Tearing down.."                                                                                                                             
                                                     [INFO  netavark::firewall] Using iptables firewall driver
                                                                                                              [DEBUG netavark::commands::teardown] Setting up network podman with driver bridge
                                                                                                                                                                                               [DEBUG netavark::network::core_utils] bridge has 1 connected interfaces
                                                                                                                                                                                                                                                                      [DEBUG netavark::network::core] Container veth name being removed: "eth0"
                                         [DEBUG netavark::network::core] Container veth removed: "eth0"
                                                                                                       [DEBUG netavark::commands::teardown] "Teardown complete"
                                                                                                                                                               DEBU[0000] Unmounted container "b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65" 
DEBU[0000] Removing container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65                                                                                                                                                                   
DEBU[0000] Cleaning up container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
DEBU[0000] Network is already cleaned up, skipping...   
DEBU[0000] Container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 storage is already unmounted, skipping... 
DEBU[0000] Removing all exec sessions for container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 
DEBU[0000] Container b2004e4b168c51be76575b6f876b13bdccc8b877ba74471bdf269003b025af65 storage is already unmounted, skipping... 
DEBU[0000] ExitCode msg: "crun: mount `/dev/loop1` to `/mnt`: not a directory: oci runtime error" 
Error: OCI runtime error: crun: mount `/dev/loop1` to `/mnt`: Not a directory
```

**Describe the results you expected:**
Run successfully.

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18
Built:        Fri May  6 18:15:54 2022
OS/Arch:      linux/amd64

```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 96.84
    systemPercent: 1.02
    userPercent: 2.14
  cpus: 12
  distribution:
    distribution: fedora
    variant: workstation
    version: "36"
  eventLogger: journald
  hostname: fedora
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.17.8-300.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 25378160640
  memTotal: 33403359232
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 22m 47.35s
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 510389125120
  graphRootUsed: 67064430592
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.1.0
  Built: 1651853754
  BuiltTime: Fri May  6 18:15:54 2022
  GitCommit: ""
  GoVersion: go1.18
  Os: linux
  OsArch: linux/amd64
  Version: 4.1.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**


Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
```
Linux fedora 5.17.8-300.fc36.x86_64 #1 SMP PREEMPT Mon May 16 01:00:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by mheon** [comment]
Can you try the `runc` runtime instead, to see if this is a crun issue?

@giuseppe PTAL

**Comment by tupyy** [comment]
Using `runc` I got this:
```
DEBU[0000] Container b7a0fd837e66a1fe4d2a614c944b02cc002ce2ac32a2eacd061788b6c64007ec storage is already unmounted, skipping... 
DEBU[0000] ExitCode msg: "runc: runc create failed: unable to start container process: error during container init: error mounting \"/dev/loop1\" to rootfs at \"/mnt\": mount /dev/loop1:/mnt (via /proc/self/fd/8), flags: 0x5002: not a directory: oci runtime error" 
Error: OCI runtime error: runc: runc create failed: unable to start container process: error during container init: error mounting "/dev/loop1" to rootfs at "/mnt": mount /dev/loop1:/mnt (via /proc/self/fd/8), flags: 0x5002: not a directory
```

**Comment by rhatdan** [comment]
If you mount -v /mnt:/mnt does it work?

**Comment by tupyy** [comment]
Yes. It works.

**Comment by rhatdan** [comment]
So this is something to do with a block device being added to a container.

Could you try to mount the block device on a subdir of /mnt?

--device /dev/loop1:/mnt/loop1

If that works, then it might be something about mounting a device on a directory?  If not, then maybe we have difficulty creating block devices.

**Comment by tupyy** [comment]
When I'm running: 
``` podman run -it --rm --mount=type=bind,src=/dev/loop1,dst=/mnt/loop1 --log-level=trace --security-opt label=disable fedora:35 /bin/bash
```
it seems to work but the device is not mounted. Only exposed:
```
DEBU[0000] running conmon: /usr/bin/conmon               args="[--api-version 1 -c a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a -u a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a -r /usr/bin/crun -b /var/lib/containers/storage/overlay-containers/a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a/userdata -p /run/containers/storage/overlay-containers/a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a/userdata/pidfile -n intelligent_lumiere --exit-dir /run/libpod/exits --full-attach -s -l journald --log-level trace --runtime-arg --log-format=json --runtime-arg --log --runtime-arg=/run/containers/storage/overlay-containers/a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a/userdata/oci-log -t --conmon-pidfile /run/containers/storage/overlay-containers/a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /var/lib/containers/storage --exit-command-arg --runroot --exit-command-arg /run/containers/storage --exit-command-arg --log-level --exit-command-arg trace --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/libpod --exit-command-arg --network-config-dir --exit-command-arg  --exit-command-arg --network-backend --exit-command-arg netavark --exit-command-arg --volumepath --exit-command-arg /var/lib/containers/storage/volumes --exit-command-arg --runtime --exit-command-arg crun --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg --syslog --exit-command-arg container --exit-command-arg cleanup --exit-command-arg --rm --exit-command-arg a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a]"
INFO[0000] Running conmon under slice machine.slice and unitName libpod-conmon-a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a.scope 
DEBU[0000] Received: 970349                                                                                                                           
INFO[0000] Got Conmon PID as 970343                     
DEBU[0000] Created container a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a in OCI runtime 
DEBU[0000] Attaching to container a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a           
DEBU[0000] Starting container a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a with command [/bin/bash] 
DEBU[0000] Received a resize event: {Width:294 Height:78}                                                               
DEBU[0000] Started container a53799adcc8dded934887c3c778a4afdd1437fcf86c956b8cd86850ce510a98a 
INFO[0000] Received shutdown.Stop(), terminating!        PID=970213                           
DEBU[0000] Enabling signal proxying                                
[root@a53799adcc8d /]# cd /mnt
[root@a53799adcc8d mnt]# ls
loop1
[root@a53799adcc8d mnt]# ls -al
total 0
drwxr-xr-x 1 root root   10 Jun  3 13:21 .
dr-xr-xr-x 1 root root   12 Jun  3 13:21 ..
brw-rw---- 1 root disk 7, 1 Jun  3 07:01 loop1
[root@a53799adcc8d mnt]# 
```

**Comment by rhatdan** [comment]
Yes that is correct, it should exists by not be mounted inside of the container, You would need to mount it within the container.
@giuseppe @AkihiroSuda Should runtimes fail if user attempts to mount a device on a  directory inode?

**Comment by tupyy** [comment]
Hello,
After some digging in `runc`, I've found that the mount is done by a syscall:
https://github.com/opencontainers/runc/blob/a7a45d7d2721314ea325874c2adc69723f6b7162/libcontainer/mount_linux.go#L57

Now, I tried to achieve the same thing with this code:
```go
package main

import (
	"fmt"

	"golang.org/x/sys/unix"
)

func main() {
	var flags int
	flags |= unix.MS_BIND | unix.MS_RDONLY
	err := unix.Mount("/dev/loop1", "/mnt", "bind", uintptr(flags), "")

	fmt.Println(err)
}
```
which gives the same error. If I'm changing `bind` by `ext4` it works but in podman there is no such type `ext4`.
So, I guess this is the right behavior. We cannot mount block devices with `mount` option.

Am I missing something here?
